src/Features/LanguageServer/ProtocolUnitTests/CodeActions/CodeActionsTests.cs

```diff
@@ -62,15 +101,31 @@ private static LSP.CodeActionParams CreateCodeActionParams(LSP.Location caret)
                 }
             };
 
-        private static LSP.Command CreateCommand(string title, LSP.Location location)
-            => new LSP.Command()
+        internal static LSP.VSCodeAction CreateCodeAction(
+            string title, LSP.CodeActionKind kind, LSP.VSCodeAction[] children,
+            CodeActionResolveData data, LSP.Diagnostic[] diagnostics,
+            LSP.WorkspaceEdit edit = null, LSP.Command command = null)
+        {
+            var action = new LSP.VSCodeAction
             {
                 Title = title,
-                CommandIdentifier = "Roslyn.RunCodeAction",
-                Arguments = new object[]
-                {
-                    CreateRunCodeActionParams(title, location)
-                }
+                Kind = kind,
+                Children = children,
+                Data = data,
+                Diagnostics = diagnostics,
             };
+
+            if (edit != null)
```

Comment:
can't we just do action.Edit = edit?  if it's null it's no different from the default value